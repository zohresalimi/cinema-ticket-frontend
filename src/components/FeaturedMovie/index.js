import React, { useState } from "react";
import ReactPlayer from "react-player/youtube";
import {
  Wrapper,
  FullImage,
  Image,
  ShadowBg,
  Info,
  Tag,
  MovieTitle,
  Link,
} from "./style";
import PlayButton from "../PlayButton";
import Modal from "../Modal";

function FeaturedMovie() {
  const [playVideo, setPlayVideo] = useState(false);
  const [modalOpen, setModalOpen] = useState(false);

  const toggleModalState = () => {
    setModalOpen(!modalOpen);
  };
  return (
    <Wrapper>
      <FullImage>
        <Image
          alt=""
          src="https://catalog.cinema-api.com/cf/images/ncg-images/9e5d23a9c1624f68859caedbf35ec3de.jpg?width=1920&version=9CD85BE7BA3CC619CC4E8D5A82DCA1C1&format=webp"
        />
        <ShadowBg />
        {playVideo && (
          <Modal setPlayVideo={setPlayVideo}>
            <div className="player-wrapper">
              <ReactPlayer
                className="react-player"
                url="https://youtu.be/tiBE56vQ0Fg"
                config={{
                  youtube: {
                    playerVars: { modestbranding: 1, showinfo: 0, controls: 0 },
                  },
                }}
              />
            </div>
          </Modal>
        )}
        <PlayButton setPlayVideo={setPlayVideo} />

        <Info>
          <Tag> Movie</Tag>
          <MovieTitle>
            <Link href="/">Movie Tilte</Link>
          </MovieTitle>
        </Info>
      </FullImage>
    </Wrapper>
  );
}

export default FeaturedMovie;
