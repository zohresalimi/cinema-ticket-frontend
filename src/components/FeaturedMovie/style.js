import styled from "styled-components";

export const Div = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: inherit;
  align-items: center;
`;

export const Wrapper = styled.div`
  display: flex;
  align-items: center;
  width: 100%;
  background-color: ${(props) => props.theme.colors.black};
`;

export const FullImage = styled.div`
  height: ${(props) => props.theme.fullImage.height};
  width: 100%;
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  box-sizing: border-box;
  overflow: hidden;

  @media (max-width: 375px) {
    max-height: 540px;
  }
`;

export const Link = styled.a`
  display: inline-block;
  vertical-align: top;
  cursor: pointer;
  font-weight: ${(props) => props.weight};
  &:hover {
    text-decoration: underline;
  }
`;
export const Image = styled.img`
  width: 100%;
  max-width: 100%;
  max-height: 100%;
  object-fit: cover;
  object-position: bottom;
  height: 100%;
`;

export const ShadowBg = styled.div`
  background: linear-gradient(
    -180deg,
    transparent 50%,
    rgba(0, 0, 0, 0.68) 85%,
    #000
  );
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
`;

export const Info = styled.div`
  position: absolute;
  width: 100%;
  max-width: 48rem;
  left: 50%;
  bottom: 2rem;
  padding: 0 1rem;
  text-align: center;
  white-space: normal;
  transform: translateX(-50%);
`;

export const Tag = styled.span`
  font-family: ${(props) => props.theme.fontFamily.sf};
  font-size: ${(props) => props.theme.fontSizes.tag};
  background: ${(props) => props.theme.colors.red};
  color: ${(props) => props.theme.colors.white};
  border-radius: 1px;
  line-height: 1.5rem;
  height: 1.5rem;
  padding: 0.3rem 0.6rem;
`;
export const MovieTitle = styled.h1`
  font-family: ${(props) => props.theme.fontFamily.sf};
  font-size: ${(props) => props.theme.fontSizes.large};
  color: ${(props) => props.theme.colors.white};
  line-height: 3.5rem;
  text-transform: capitalize;
  margin: 20px auto;
`;
