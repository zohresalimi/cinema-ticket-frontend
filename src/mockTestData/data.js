import React from "react";
import { Router } from "@reach/router";
import AppContext from "../store/context";
import Theme from "../Styles/Theme";

export const getTestStore = () => {
  return {
    dispatch: () => {},
    state: {
      user: {
        name: "",
        email: "",
      },
      movies: {
        children: [
          {
            _id: "5fa71823adc463397a50c501",
            actors: ["Holland Roden", "Ronen Rubinstein", "Keegan Allen"],
            age: "7",
            coverImage:
              "https://catalog.cinema-api.com/cf/images/ncg-images/68bfeef55ada4e7ab0606db2af0f40ad.jpg?width=240&version=EA2E80EE161E5DA161C6E26487054C9F&format=webp",
            description:
              "Poppy and Kvist embark on an adventure that will take them far beyond",
            director: "aaron woodley",
            duration: "1:31",
            genre: ["family", "children"],
            images: [],
            largeImage:
              "https://catalog.cinema-api.com/cf/images/ncg-images/79276cead82341d2a004f8e8f99a88e9.jpg?width=1920&version=08F0C83177D59FB94C4C809F63F67E2A&format=webp",
            name: "trolls 2: world tour",
            originalTitle: "swedish",
            premiere: "2020-10-03T15:18:52.722Z",
            rooms: ["5fa653d0a7f1201999dd1ac0"],
            trailerUrl: "https://youtu.be/P9A6hOg9QQ0",
          },
        ],
        premiered: [
          {
            _id: "5fa71823adc463397a50c502",
            actors: ["Holland Roden", "Ronen Rubinstein", "Keegan Allen"],
            age: "7",
            coverImage:
              "https://catalog.cinema-api.com/cf/images/ncg-images/68bfeef55ada4e7ab0606db2af0f40ad.jpg?width=240&version=EA2E80EE161E5DA161C6E26487054C9F&format=webp",
            description:
              "Poppy and Kvist embark on an adventure that will take them far beyond",
            director: "aaron woodley",
            duration: "1:31",
            genre: ["family", "children"],
            images: [],
            largeImage:
              "https://catalog.cinema-api.com/cf/images/ncg-images/79276cead82341d2a004f8e8f99a88e9.jpg?width=1920&version=08F0C83177D59FB94C4C809F63F67E2A&format=webp",
            name: "follow me",
            originalTitle: "swedish",
            premiere: "2020-10-05T15:18:52.722Z",
            rooms: ["5fa653c0643c06586bba7d78"],
            trailerUrl: "https://youtu.be/P9A6hOg9QQ0",
          },
        ],
        upcoming: [
          {
            _id: "5fa71823adc463397a50c503",
            actors: ["Holland Roden", "Ronen Rubinstein", "Keegan Allen"],
            age: "7",
            coverImage:
              "https://catalog.cinema-api.com/cf/images/ncg-images/68bfeef55ada4e7ab0606db2af0f40ad.jpg?width=240&version=EA2E80EE161E5DA161C6E26487054C9F&format=webp",
            description:
              "Poppy and Kvist embark on an adventure that will take them far beyond",
            director: "aaron woodley",
            duration: "1:31",
            genre: ["family", "children"],
            images: [],
            largeImage:
              "https://catalog.cinema-api.com/cf/images/ncg-images/79276cead82341d2a004f8e8f99a88e9.jpg?width=1920&version=08F0C83177D59FB94C4C809F63F67E2A&format=webp",
            name: "honest thief",
            originalTitle: "swedish",
            premiere: "2021-01-03T15:18:52.722Z",
            rooms: ["5fa653c0643c06586bba7d78"],
            trailerUrl: "https://youtu.be/P9A6hOg9QQ0",
          },
        ],
        allMovies: {
          "5fa71823adc463397a50c501": "children",
          "5fa71823adc463397a50c502": "premiered",
          "5fa71823adc463397a50c503": "upcoming",
        },
      },
      cinemas: [
        {
          _id: "5fa64fe234fbbf9945f3a3ce",
          name: "filmstaden kista",
          purchaseEndTime: "21:30",
          purchaseStartTime: "8:00",
          rooms: ["5fa653c0643c06586bba7d78", "5fa653d0a7f1201999dd1ac0"],
        },
      ],
      ticket: {
        movie: {},
        status: false,
        quantity: 0,
        price: 0,
        seatNumbers: new Map(),
        user: {},
        showing: {},
      },
      showings: [
        {
          _id: "5fbe9bd31f58462b5c759b5c",
          capacity: 4,
          cinema: "5fa64fe234fbbf9945f3a3ce",
          endTime: null,
          movie: {
            _id: "5fa71823adc463397a50c502",
            actors: ["Holland Roden", "Ronen Rubinstein", "Keegan Allen"],
            age: "7",
            description:
              "Poppy and Kvist embark on an adventure that will take them far beyond",
            director: "aaron woodley",
            duration: "1:31",
            genre: ["family", "children"],
            images: [],
            name: "follow me",
            originalTitle: "swedish",
            premiere: "2020-10-05T15:18:52.722Z",
            rooms: ["5fa653c0643c06586bba7d78"],
            trailerUrl: "https://youtu.be/P9A6hOg9QQ0",
          },
          price: 100,
          room: { _id: "5fa653c0643c06586bba7d78" },

          seats: [
            [
              { seatNumber: 1, taken: "available" },
              { seatNumber: 1, taken: "available" },
            ],
            [
              { seatNumber: 1, taken: "available" },
              { seatNumber: 1, taken: "available" },
            ],
          ],
          startTime: "2020-11-13T16:00:00.000Z",
        },
      ],
    },
  };
};

export const WithProvider = (props) => {
  const { state, dispatch } = getTestStore();
  return (
    <Theme>
      <AppContext.Provider
        value={{ state: props.defaultValue || state, dispatch }}
      >
        <Router>{props.children}</Router>
      </AppContext.Provider>
    </Theme>
  );
};
