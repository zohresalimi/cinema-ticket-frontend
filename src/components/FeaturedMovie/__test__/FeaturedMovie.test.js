import * as React from "react";
import { render, act, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import { getTestStore, WithProvider } from "../../../mockTestData/data";
import FeaturedMovie from "../";

jest.mock("react-i18next", () => ({
  useTranslation: () => [(key) => key, { changeLanguage: jest.fn() }],
}));

function renderWrapper(props) {
  let component;

  act(() => {
    component = render(
      <WithProvider>
        <FeaturedMovie {...props} path="/" />
      </WithProvider>
    );
  });

  return component;
}

describe("FeaturedMovie Component Testing", () => {
  afterAll(() => {
    jest.resetAllMocks();
  });

  test("take snapshot", async () => {
    const { container } = await renderWrapper();
    expect(container.firstChild).toMatchSnapshot();
  });

  test("should play video", async () => {
    const setState = jest.fn();
    jest.spyOn(React, "useState").mockImplementation(() => [true, setState]);
    const movie = getTestStore().state.movies.premiered[0];

    const { container } = await renderWrapper({
      movie,
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
