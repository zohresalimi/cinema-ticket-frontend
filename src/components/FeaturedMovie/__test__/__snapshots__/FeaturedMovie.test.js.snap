// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedMovie Component Testing should play video 1`] = `
<div
  style="outline: none;"
  tabindex="-1"
>
  <div
    class="style__Wrapper-sc-16suol2-1 NzduN"
  >
    <div
      class="style__FullImage-sc-16suol2-2 ezesCl"
    >
      <img
        alt=""
        class="style__Image-sc-16suol2-4 krudRE"
      />
      <div
        class="style__ShadowBg-sc-16suol2-5 lmCCjh"
      />
      <div
        class="full-page"
      >
        <div
          class="video-wrapper"
        >
          <div
            class="close-btn"
            data-testid="close-btn"
            role="button"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-times-circle fa-w-16 "
              data-icon="times-circle"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
                fill="currentColor"
              />
            </svg>
          </div>
          <div
            class="player-wrapper"
          >
            <div
              class="react-player"
              style="width: 100%; height: 100%;"
            />
          </div>
        </div>
      </div>
      <div
        class="style__Div-y1of7i-0 kOhgOG"
        data-testid="play-btn"
      >
        <div
          class="style__RoundBut-y1of7i-2 iqNHdy"
        >
          <a
            class="style__PlayBtn-y1of7i-1 BDDhv"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-play fa-w-14 "
              data-icon="play"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z"
                fill="currentColor"
              />
            </svg>
          </a>
        </div>
      </div>
      <div
        class="style__Info-sc-16suol2-6 kmYNvI"
      >
        <span
          class="style__Tag-sc-16suol2-7 feJZOr"
        >
          at the cinema now
        </span>
        <h1
          class="style__MovieTitle-sc-16suol2-8 fAYStm"
        >
          <p />
        </h1>
      </div>
    </div>
  </div>
</div>
`;

exports[`FeaturedMovie Component Testing take snapshot 1`] = `
<div
  style="outline: none;"
  tabindex="-1"
>
  <div
    class="style__Wrapper-sc-16suol2-1 NzduN"
  >
    <div
      class="style__FullImage-sc-16suol2-2 ezesCl"
    >
      <img
        alt=""
        class="style__Image-sc-16suol2-4 krudRE"
        src="https://catalog.cinema-api.com/cf/images/ncg-images/79276cead82341d2a004f8e8f99a88e9.jpg?width=1920&version=08F0C83177D59FB94C4C809F63F67E2A&format=webp"
      />
      <div
        class="style__ShadowBg-sc-16suol2-5 lmCCjh"
      />
      <div
        class="style__Div-y1of7i-0 kOhgOG"
        data-testid="play-btn"
      >
        <div
          class="style__RoundBut-y1of7i-2 iqNHdy"
        >
          <a
            class="style__PlayBtn-y1of7i-1 BDDhv"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-play fa-w-14 "
              data-icon="play"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z"
                fill="currentColor"
              />
            </svg>
          </a>
        </div>
      </div>
      <div
        class="style__Info-sc-16suol2-6 kmYNvI"
      >
        <span
          class="style__Tag-sc-16suol2-7 feJZOr"
        >
          at the cinema now
        </span>
        <h1
          class="style__MovieTitle-sc-16suol2-8 fAYStm"
        >
          <p>
            follow me
          </p>
        </h1>
      </div>
    </div>
  </div>
</div>
`;
